src/rust/engine/src/nodes.rs

```diff
@@ -1068,6 +1068,9 @@ impl WrappedNode for Task {
       .then(move |task_result| match task_result {
         Ok(val) => match externs::get_type_for(&val) {
           t if t == context.core.types.generator => Self::generate(context, params, entry, val),
```

Comment:
Nit: this could be `if t == context.core.types.generator || t == context.core.types.coroutine`.